server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -294,7 +304,7 @@ void ensureConnected(String clusterAlias, ActionListener<Void> listener) {
      * Returns whether the cluster identified by the provided alias is configured to be skipped when unavailable
      */
     public boolean isSkipUnavailable(String clusterAlias) {
-        return getRemoteClusterConnection(clusterAlias).isSkipUnavailable();
+        return isStateless || getRemoteClusterConnection(clusterAlias).isSkipUnavailable();
```

Comment:
Nit: we can keep this method unchanged since we guarantee its value to be correct in other places. We can add an assertion here if necessary.